src/cephadm/cephadm

```diff
@@ -3300,6 +3300,16 @@ def deploy_daemon_units(
                     bind_mounts=get_container_binds(ctx, fsid, daemon_type, daemon_id),
                     cname='ceph-%s-%s.%s-activate' % (fsid, daemon_type, daemon_id),
                 )
+                if 'cluster' in ctx and ctx.cluster:
+                    # ctx.cluster is only set during adoption of a daemon from a cluster
+                    # with a custom name (not "ceph"). The initial activate command the first
+                    # time we start the new cephadm based systemd unit for this osd must account
+                    # for this by mounting to the correct data dir in the container. Otherwise
+                    # necessary files from the old data dir of the daemon won't be copied over
+                    # to the new data dir on the host. After the first start (e.g. on any redeploys)
+                    # this is no longer necessary as we will have these files in the data dir on the host
+                    if data_dir in prestart.volume_mounts:
+                        prestart.volume_mounts[data_dir] = f'/var/lib/ceph/osd/{ctx.cluster}-{daemon_id}'
```

Comment:
consider using `ctx.data_dir` instead of `/var/lib/ceph`

Reply:
the reason for using the hardcoded value here is that this isn't actually referring to our data dir. It's actually the spot in the container where this data needs to be located which is always going to be /var/lib/ceph/osd/{cluster-name}-{osd-id}. It corresponds to https://github.com/ceph/ceph/blob/master/src/cephadm/cephadm#L2786

Reply:
Ohh, I see. Actually there'a  lot of these hard-coded refs in the binary. Plz, ignore my comment.